azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/handlers/CallStateHandler.kt

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.handlers
+
+import com.azure.android.communication.ui.calling.configuration.CallCompositeConfiguration
+import com.azure.android.communication.ui.calling.models.CallCompositeCallState
+import com.azure.android.communication.ui.calling.redux.Store
+import com.azure.android.communication.ui.calling.redux.state.CallingStatus
+import com.azure.android.communication.ui.calling.redux.state.ReduxState
+import kotlinx.coroutines.flow.collect
+
+internal class CallStateHandler(
+    private val configuration: CallCompositeConfiguration,
+    private val store: Store<ReduxState>,
+) {
+    private var callingStatus: CallingStatus? = null
```

Comment:
Please use stateFlow to hold status. it will ensure thread-safety

Reply:
As discussed will not cause threading issue for current code. We can still add this part of code in case future start is executed on different threads.